### PR TITLE
fix(connector): send join requests in a single batch

### LINKED ISSUE #112: Send all the join requests in a single batch
> RDP 4.0, 5.0, 5.1, 5.2, 6.0, 6.1, 7.0, 7.1, 8.0, 10.2, 10.3, 10.4, and 10.5 clients send a Channel Join Request to the server only after the Channel Join Confirm for a previously sent request has been received. RDP 8.1, 10.0, and 10.1 clients send all of the Channel Join Requests to the server in a single batch to minimize the overall connection sequence time.

Let’s send all the Join Requests in a single batch in IronRDP.

[TODO comment in code](https://github.com/Devolutions/IronRDP/blob/ccdd3825b08c64d7736dd3207c3d7281d0445b30/crates/connector/src/channel_connection.rs#L139)

### PR DESCRIPTION
**Pull request by CBenoit**
Closes #112
Related to #314

### CODE REVIEW

File: crates/ironrdp-connector/src/channel_connection.rs [hunk 1]
```diff
@@ -125,49 +126,46 @@ impl Sequence for ChannelConnectionSequence {
...
+                    let channel_join_request = mcs::ChannelJoinRequest {
+                        initiator_id: user_channel_id,
+                        channel_id,
+                    };
 
-                debug!(message = ?channel_join_request, "Send");
+                    debug!(message = ?channel_join_request, "Send");
 
-                let written = ironrdp_pdu::encode_buf(&channel_join_request, output).map_err(ConnectorError::pdu)?;
+                    written += ironrdp_pdu::encode_buf(&channel_join_request, output).map_err(ConnectorError::pdu)?;
```
Comment (elmarco) [inline crates/ironrdp-connector/src/channel_connection.rs:159]: written = written.checked_add(ironrdp_pdu::encode_buf(&channel_join_request, output).map_err(ConnectorError::pdu)?).unwrap();

Comment (elmarco) [reply crates/ironrdp-connector/src/channel_connection.rs:159]: lgtm otherwise

Comment (CBenoit) [reply crates/ironrdp-connector/src/channel_connection.rs]: Thank you!

**Review by pacmancoder — APPROVED** [review]
LGTM

**Pull request merged**